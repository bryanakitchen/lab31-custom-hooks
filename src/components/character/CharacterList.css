.plain {
    background: white;
    color: black;
}

.style {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    
    li {
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
        justify-content: center;
        align-items: center;
        border-radius: 10px;
        margin: 5px;
        width: 220px;
        height: 220px;
        background: #F07167;
    }
}