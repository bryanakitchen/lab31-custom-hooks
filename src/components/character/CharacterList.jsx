import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import { Link } from 'react-router-dom';

function CharacterList({ characters }) {

  const characterElements = characters.map(character => (
    <li key={character.id}>
      <Link to={`/character/${character.id}`}>
        <Character name={character.name} image={character.image} /> 
      </Link>
    </li>
  ));
  
  return (
    <ul data-testid="characters">
      {characterElements}
    </ul>
  );
}

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }))
};

export default CharacterList;
