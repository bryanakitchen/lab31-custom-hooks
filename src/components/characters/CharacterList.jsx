import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

const CharacterList = ({ characters }) => {
  const characterElements = characters.map(character => (
    <li key={character.id}>
      <Character name={character.name} image={character.image} />
    </li>
  ));
  
  return (
    <ul data-testid="characters">
      {characterElements}
    </ul>
  );
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }))
};

export default CharacterList;
