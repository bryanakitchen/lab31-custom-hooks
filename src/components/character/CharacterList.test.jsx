/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList 
          characters={[
            {
              id: '5f5fb4bbbfd05c2aed82e460',
              name: 'Admiral',
              image: 'https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138'
            },
            {
              id: '5f5fb4bbbfd05c2aed82e462',
              name: 'Agnes',
              image: 'https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432'
            }
          ]} />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
