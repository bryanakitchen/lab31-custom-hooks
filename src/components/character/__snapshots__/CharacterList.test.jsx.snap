// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <ul
    class="plain"
    data-testid="characters"
  >
    <li>
      <a
        href="/character/5f5fb4bbbfd05c2aed82e460"
        style="text-decoration: none;"
      >
        <figure>
          <img
            alt="Admiral"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
            style="width: 100px;"
          />
          <figcaption>
            Admiral
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/character/5f5fb4bbbfd05c2aed82e462"
        style="text-decoration: none;"
      >
        <figure>
          <img
            alt="Agnes"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
            style="width: 100px;"
          />
          <figcaption>
            Agnes
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
